proposals/0016-simd-program-rebatable-account-write-fees.md

```diff
@@ -0,0 +1,621 @@
+to quote quite often. The goal of this proposal is to reduce spamming without
+really increasing fees for everyone. Encourage users to create proper
+transactions. Read the cluster state before creating transactions instead of
+spamming. We are motivated to improve the user experience, and user activity and
+keep Solana a low gas fee blockchain. Keeping gas fees low and increasing user
+activity on the chain will help all the communities based on Solana grow. This
+will also help users to gain more control over accounts owned by them.
+
+Note that this proposal does not aim to increase gas fees for all transactions
+or add any kind of congestion control mechanism.
+
+## Alternatives Considered
+
+* Having a fixed write lock fee and a read lock fee.
+
+Pros: Simpler to implement, Simple to calculate fees.
+
+Cons: Will increase fees for everyone, dapp cannot decide to rebate fees, (or
+not all existing apps have to develop a rebate system). Fees cannot be decided
+by the dapp developer. Penalizes everyone for a few bad actors.
+
+* Passing fees for each account by instruction and validating them
+  during the execution inside user application code
+
+Pros: High efficiency, minimal performance impact, more generic, no need to
+change the account structure.
+
+Cons: Cannot prevent denial of service attacks, or account blocking
+attacks that do not call into the respective program
+
+## New Terminology
+
+Program Rebatable Account Write Fees or PRAW Fees: Fees that are decided by the
+dapp developer will be charged if a user wants to use the dapp. They will be
+applied even if the transaction fails contrary to lamport transfers. The program
+can decide to rebate these fees back to the user if certain conditions decided
+by dapp developers are met.
+
+### Other terminology
+
+Account `Authority`: The true authority for the account. Let's take an example
+of associated token account. A associated token account is a `PDA` derived from
+associated token program, where the owner is token program. But the program
+internally saves who has the authority over the account. Only token program can
+change the data of a token account. For operations like withdrawing tokens, the
+authority of the associated token account has to be the signer for the
+transaction containing transfer instruction. To receive tokens from another
+account, the authority's signature is not required.
+
+`Other Fees`: Fees other than PRAW fees `base fees + prioritization fees`.
+
+## Detailed Design
+
+PRAW fees enable dapp developer to decide who to rebate and who to penalize
+through runtime features. In particular, programs needs to be able to penalize,
+even if the application is not CPI'd to.
+
+The checks on the PRAW fees will be taken care of by Solana runtime. Total PRAW
+fees paid will be included in the transaction through a special
+`SetProgramRebatableAccountWriteFees` similar to the `ComputeBudget` program so
```

Comment:
mention the new program name & instruction and clarify that SetProgramRebatableAccountWriteFees is a TVL

Reply:
done.